Implemented YouTube search.

leonplondon opened the pull request:
# Summary
Youtube search implementation.
Added dependencies.
Markup of UI.
Perform request to youtube using YouTube provided services.
Implemented infinite scroll.

# Trello Card
https://trello.com/c/oe0lUsLj/37-youtube-search

# Screenshots
![screenshot_1544220388](https://user-images.githubusercontent.com/42459619/49675177-8de23d80-fa42-11e8-9dd4-6c33daa135e3.png)

leonplondon committed:
Implemented detail view for news

leonplondon committed:
Implemented detail view for news

leonplondon committed:
Implemented detail view for news

leonplondon committed:
Implemented detail view for news

leonplondon committed:
Fixed conflicts

leonplondon committed:
Removed duplicated dimens

leonplondon committed:
Fixed some bad practices/style

leonplondon committed:
Fixed some code style

leonplondon committed:
Added Google Auth dependencies through Google Play services.
Implemented logic to perform Google Sign In.
Added required strings to inform user about error during the process.

leonplondon committed:
Fixed PR#8

leonplondon committed:
Fixed PR#9.
Format strings to present to user.

leonplondon committed:
Implemented YouTube search.
Added libraries.
Implemented business logic.
Implemented related UI.

leonplondon committed:
Implemented YouTube search.
Added libraries.
Implemented business logic.
Implemented related UI.

leonplondon committed:
Fixed PR#8

leonplondon committed:
Merge pull request #9 from wolox-training/login-google

Google Login

leonplondon committed:
Changed identation of NewsPresenter

leonplondon committed:
Merge branch 'development' into news-watch

dylan-muszel commented on app/build.gradle:
```diff
@@ -28,6 +28,8 @@ android {
         versionCode 1
         versionName "1.0.0"
 
+        multiDexEnabled true
```
What is Multidex for?

leonplondon replied:
When the limit of 64k methods is reached on Android, the application can't be compiled as a single classes.dex bytecode, in order to fix the problem, multidex lib+configuration is required. The problem occurs when adding multiples libraries with a large method count as dependency.

leonplondon replied:
More info here --> [Multidex](https://developer.android.com/studio/build/multidex)

dylan-muszel commented on app/src/main/java/ar/com/wolox/android/example/model/SupportedYouTubeObject.kt:
```diff
@@ -0,0 +1,8 @@
+/**
+ * Model to wrap a [com.google.api.services.youtube.model.SearchResult] to pass ids as extras
+ */
+data class SupportedYouTubeObject(val kind: String, val playlistId: String?, val videoId: String?) : Serializable
```
I don't like this name, this doesn't express what this object is for. I think something like `SearchResultAdapter` could be better, but feel free to think another 😄

leonplondon replied:
This model works as a `com.google.api.services.youtube.model.SearchResult` subset to pass only supported `YouTube API` responses, I'll name it as `SearchItemAdapter`.

leonplondon replied:
As the class is a model, I find `Adapter` not appropriate, so I think I'll name that model as `SearchResultModelAdapter` to give it more sense.

dylan-muszel commented on app/build.gradle:
```diff
@@ -142,4 +144,17 @@ dependencies {
+    implementation files('libs/YouTubeAndroidPlayerApi.jar')
+
+    // Multidex
+    compile 'com.android.support:multidex:1.0.3'
```
`compile` is obsolete, use `implementation` instead.

leonplondon replied:
My bad, I'll change it.

dylan-muszel commented on app/src/main/java/ar/com/wolox/android/example/ui/youtube/YouTubeSearchPresenter.kt:
```diff
@@ -0,0 +1,76 @@
+        Thread {
+            try {
+                val result = search.execute()
+                Log.d("ldpenal", "here'")
```
Remove this `Log`

leonplondon replied:
Of course

dylan-muszel reviewed (changes requested):
It's a good work, but there are some questions.

leonplondon committed:
Merge pull request #8 from wolox-training/news-watch

News detail frontend + backend

leonplondon committed:
Fixes for PR#10.
Renamed model to adapt YouTube's search result to intent extra.
Removed typo adding dependency.

leonplondon committed:
Separated strings for unexpected error, one with format the other without it.

leonplondon committed:
Fixed merge conflicts

leonplondon committed:
Removed unused code file

dylan-muszel reviewed (approved):
Good work! I don't know if @lmiotti wants to add a review, but for me it's approved.

![](https://media1.giphy.com/media/l0MYDGA3Du1hBR4xG/giphy.gif)

lmiotti commented on app/src/main/res/layout/item_search_result.xml:
```diff
@@ -0,0 +1,70 @@
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@id/mVideoImage"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec diam ligula, vulputate eu lectus nec, rutrum vulputate neque. Nulla facilisi. Nullam id finibus lorem, at ornare neque." />
```
Add this text to strings file

lmiotti commented on app/src/main/res/layout/item_search_result.xml:
```diff
@@ -0,0 +1,70 @@
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="@id/mVideoTitleText"
+            app:layout_constraintTop_toBottomOf="@id/mVideoTitleText"
+            tools:text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec diam ligula, vulputate eu lectus nec, rutrum vulputate neque. Nulla facilisi. Nullam id finibus lorem, at ornare neque." />
```
Some as above

lmiotti reviewed (changes requested):
This is quite good. Correct the small things I've suggested and we're done

leonplondon committed:
Fixed PR#10.
Extracted strings used as placeholder for text base views.

lmiotti reviewed (approved):
Keep on like this

![image](https://user-images.githubusercontent.com/36635668/49952039-0e3cee80-feda-11e8-8edc-114e3d6ea721.png)
